File: app/src/main/java/com/ikvych/cocktail/repository/DrinkApiRepository.java
package com.ikvych.cocktail.repository;

import androidx.lifecycle.MutableLiveData;

import com.ikvych.cocktail.model.Drink;

import java.util.List;

public interface DrinkApiRepository {

    void updateDrinksLiveData(String name);

    MutableLiveData<List<Drink>> getDrinksLiveData();

    List<Drink> getCurrentData();
}

File: app/src/main/java/com/ikvych/cocktail/data/network/DrinkApiRepository.kt
package com.ikvych.cocktail.data.network

import androidx.lifecycle.MutableLiveData
import com.ikvych.cocktail.data.network.model.Drink
import com.ikvych.cocktail.data.repository.source.base.BaseRepository

interface DrinkApiRepository : BaseRepository {

    // Methods for work with Api
    fun getDrinkApiLiveData(): MutableLiveData<List<Drink>>
    fun updateDrinkApiLiveData(query: String)

}